### PR TITLE
Add percentage letter function

### LINKED ISSUE #28: Percentage of Letter in String
# Percentage of Letter in String

Given a string `s` and a character `letter`, return the **_percentage_** of characters in `s` that equal `letter` **_rounded down_** to the nearest whole percent.

## **Examples:**     

* **Example 1:**

    * **Input:** ` s = "foobar", letter = "o"`
    * **Output:**`33`
    * **Explanation:**
`The percentage of characters in s that equal the letter 'o' is 2 / 6 * 100% = 33% when rounded down, so we return 33.`

* **Example 2:**

   * **Input:** `s = "jjjj", letter = "k"`
  * **Output:** `0`
  * **Explanation:**
`The percentage of characters in s that equal the letter 'k' is 0%, so we return 0.`

## **Constraints:**

`1 <= s.length <= 100`
`s` consists of lowercase English letters.
`letter` is a lowercase English letter.

### PR DESCRIPTION
**Pull request by heba3**
### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by CaesarGhazi — COMMENTED** [review]
OHMYGOOOD!!! HEBAAA, you've made my life so much easier.
i don't even know what to do.

File: solutions/percentage_letter.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        >>> percentage_letter("hello", "l")
+        40
+        >>> percentage_letter("world", "w")
+        20
+        >>> percentage_letter("calculations", "z")
+        0
+"""
+
+
+def percentage_letter(s: str, letter: str) -> int:
```
Comment (CaesarGhazi) [inline solutions/percentage_letter.py]: GREAT JOB!!! 👏

Comment (CaesarGhazi) [reply solutions/percentage_letter.py]: i have only one suggestion.
which is just the variable naming.
so just the variable 's' , i'd just name it sumn else like sentence or word or text or anything you'd like really just to make the code more readable.
other than that heba, i'm pretty satisfied 😊 .

**Review by CaesarGhazi — CHANGES_REQUESTED** [review]
just the variable names, other than that it's wonderful

File: solutions/percentage_letter.py (hunk 1 above)
Comment (heba3) [reply solutions/percentage_letter.py]: @CaesarGhazi Thank you for your suggestion! I agree and will update the variable name.
I truly appreciate your feedback. Thanks again for helping improve the code!

**Comment by Elshikh-Amro** [comment]
this is amaziiiiiing

File: solutions/percentage_letter.py (hunk 1 above)
Comment (Elshikh-Amro) [reply solutions/percentage_letter.py]: i was going to mention that but goooood work everyone

**Comment by Elshikh-Amro** [comment]
i think you should define the variables more clearly in order to use snake method like 'count' can be 'count_letters'

**Comment by heba3** [comment]
@Elshikh-Amro Thank you for your feedback; I appreciate your time and suggestions. Based on your input, I’ve updated the variable name to count_letters for clarity. I typically use snake_case for multi-word names and keep single-word names simpler, but I agree this change improves readability.

**Review by CaesarGhazi — APPROVED** [review]
